wp_api/src/login.rs

```diff
@@ -98,6 +98,36 @@ pub enum OAuthResponseUrlError {
     UnsuccessfulLogin,
 }
 
+#[uniffi::export]
+pub fn create_application_password_authentication_url(
+    login_url: Arc<ParsedUrl>,
+    app_name: String,
+    app_id: Option<String>,
```

Comment:
If this is provided it needs to be a valid UUID (but not just _any_ valid UUID – it needs to be valid according to [wp_is_uuid](https://github.com/WordPress/wordpress-develop/blob/e3f28a9a11fd2c6eea7862a616738ace01d09d26/src/wp-includes/functions.php#L7021) – that basically means "a UUID, but with the right magic numbers in the right place.

Shouldn't be too hard to do in Rust, but it should probably be a programmer error to provide something invalid (and we should potentially provide an `AppId` type that's _guaranteed_ to be valid

Reply:
Addressed in https://github.com/Automattic/wordpress-rs/pull/250/commits/e6b5a05f685fd84977a7250354bb634f0284123f.